pkg/ipcache/types/types.go

```diff
@@ -63,6 +63,14 @@ func NewResourceID(kind ResourceKind, namespace, name string) ResourceID {
 	return ResourceID(str.String())
 }
 
+func (r ResourceID) Namespace() string {
+	parts := strings.SplitN(string(r), "/", 3)
```

Comment:
why n=3 ?

Reply:
ResourceID is in the form of kind/namespace/name so we split it into 3 parts and take second.
I don't see an alternative 😅 ?

Reply:
oh right, forgot about the kind part, nevermind 😅